fix(postgres): addressed trailing junk error introduced in 15.x

sutt0n opened the pull request:
when batch signing (context: using psql 15.13), i was receiving the following error:

```
ERROR:  trailing junk after parameter at or near "$3WHERE" at character
STATEMENT:  UPDATE bria_utxos
            SET spending_batch_id = $1, spending_payout_queue_id = $2, spending_sats_per_vbyte = $3WHERE account_id = $4 AND (keychain_id, tx_id, vout) IN (($5, $6, $7), ($8, $9, $10)) 
```

this just ensures there's a space between `$3` and `WHERE` in the prepared statement

edit: since postgres 15.x is being used, this occurred when 15.x came out https://git.postgresql.org/gitweb/?p=postgresql.git;a=commitdiff;h=2549f0661

edit: ci runs on my fork here https://github.com/sutt0n/bria/pull/6

sutt0n commented:
bumped `tracing-subscriber` as well, since it was failing `cargo audit` via:

```
Crate:     tracing-subscriber
Version:   0.3.18
Title:     Logging user input may result in poisoning logs with ANSI escape sequences
Date:      2025-08-29
ID:        RUSTSEC-2025-0055
URL:       https://rustsec.org/advisories/RUSTSEC-2025-0055
Solution:  Upgrade to >=0.3.20
Dependency tree:
tracing-subscriber 0.3.18
```

sutt0n commented on migrations/20250912000000_fix_uuid_nil_postgres15.up.sql:
```diff
@@ -0,0 +1,10 @@
+
+-- Patch: Ensure uuid_nil() exists for Postgres 15+ without uuid-ossp
+
+-- Patch: Always (re)define uuid_nil() for Postgres 14/15 compatibility
```
todo: remove duplicate comment. had an issue with rebasing from different remotes, this snuck past

Lakshyyaa commented:
> @Lakshyyaa seems that there's inconsistent tests failing after you retried -- are these flakey e2e tests?

idts, i have only seen tests 5 and 6 act flaky before (e2e tests)

sutt0n commented:
looking at `.e2e-logs`, i noticed this error happening quite a bit. related, perhaps?

```
FeeEstimationError - FeeEstimation: Request error: error sending request for url (https://blockstream.info/api/fee-estimates)
```

sutt0n commented:
@Lakshyyaa @bodymindarts: it seems that the e2e test suite was not hitting `mempool` cause it wasn't running (requires db) and it was defaulting to the blockstream.info API -- which i'm presuming has rate limiting or they're having issues. unsure.

regardless, i added a db in this PR for `mempool` to point to, and it seems that it's running and that e2e tests are less flaky and seem more solid. to further reduce flakiness, i've added `BATS_TEST_RETRIES=5` and a test timeout of 300s in my latest commit.

i'm able to get this passing locally (x86_64-apple-darwin), but it seems it won't pass in CI -- perhaps due to cache? i don't have enough context for the magic nix cache action to be confident in that answer. 

i realize that this e2e effort is not the initial scope of this PR, but it is a blocker and i'm trying not to spam CI. so if you'd like for me to separate this out into another PR, i can do so -- let me know.
